### ISSUE TITLE
Building the image on a system other than NixOS

### ISSUE THREAD

**Issue by DanBiscotti** [samueldr/wip-pinebook-pro#32]
Hi there,
I was trying to run the nix-build scripts (both iso and SD image) but wasn't sure which system/kernel/version it would run on. 

Initially I tried with NixOS running on a raspberry Pi (as I wanted to avoid cross-compilation and the pi was aarch64), however it failed at the kernel patch stage. I saw one of the other issues mention that it had to be a specific version of nix so I thought maybe it was that because the pi has a slightly older version of nix.

Then I tried building from Manjaro running on the pinebookpro by installing nix and using that to build however that also failed.

I'm still very new to nix and not sure if I'm missing something. I can post the errors from each try if it helps, just wanted to check if you knew if it would be possible from these to build from these systems

Thanks!

**Comment by samueldr** [comment]
Hi!

It would be easier to answer if you could copy the error messages as they are. ~~Though I will say that it's pretty much expected that the `_latest` kernel won't build as the patches haven't yet been updated.~~ ~~(EDIT: It seems to build fine with limited testing here.)~~ (Further EDIT: it fails to compile further in the build with `_latest`...)

Given [it uses `_latest` by default](https://github.com/samueldr/wip-pinebook-pro/blob/7df87f4f3baecccba79807c291b3bbd62ac61e0f/pinebook_pro.nix#L9) that may explain what you're seeing.

**Comment by samueldr** [comment]
Using `NIXPKGS_ALLOW_UNFREE=1 nix-build -A kernel_lts` on the last commit I freshly pushed, I was able to cross-compile the kernel. I **assume** it will build natively just fine with Nix. Big caveat: I have not tested the produced image, but I strongly assume that if it builds it will boots fine.

The main difference in the latest updates is that now I pin a known good Nixpkgs revision by default (so by default it will build a more and more outdated SD image).

A Nix build that fails on non-NixOS is unlikely, and would be categorized as a bug. So keep me updated on whether it works or not.

**Comment by lordcirth** [comment]
Yeah, I had to switch to the pinebook lts kernel. Note that I did first accidentally use the generic lts, which failed to boot.

**Comment by DanBiscotti** [comment]
I have just given it another go, using the latest commit caa3d3e, I tried to run `nix-build -A kernel_lts` on the pinebook itself but it failed due to out of memory. I decided to try and run it on my other x86_64 machine, which worked. So then I tried to build the sdImage which also worked, however when I boot it seems to get stuck in the bootup.
I can't tell much from the messages, it just says `INFO: task systemd-udevd:782 blocked for more than 112 seconds`.
Is this where I have to switch to the pinebook kernel?
